core/app.go

```diff
@@ -77,8 +77,7 @@ func (a *App) validateChainConditions() error {
 			return fmt.Errorf("chain with id %s does not exist", chain.ChainId)
 		}
 
-		// check sender is the valid address
-		addr, err := sdk.GetFromBech32(chain.Sender, chain.AccountPrefix)
+		ki, err := client.Keybase.Key(chain.KeyName())
```

Comment:
Why is this change needed? Would it have been simpler to keep the `key_name/sender` structure and overwrite them with correct values on infraco?

Reply:
The thing that was weird with this approach is that the config was asking you for three things that are modelled as independent but are not/do not need to be:

1. The `Key` AKA mnemonic that uniquely defines the `Sender` address, you basically need to derive that yourself from the key in the current version
2. The `KeyName` which throws an error if you attempt to `RestoreKey` to that name
3. The `Sender` address which was as above

The point here is we remove 3 bits of config, and replace them with 1 piece of config and derive everything from that.

> Would it have been simpler to keep the key_name/sender structure and overwrite them with correct values on infraco?

So not really, it would have been a load of busywork to go and compute these things first when the faucet itself could do that.